internal/cli/watch_opts.go

```diff
@@ -24,7 +24,8 @@ import (
 
 type WatchOpts struct {
 	OutputOpts
-	s *spinner.Spinner
+	s                         *spinner.Spinner
+	exponentialBackoffTrigger func(err error) bool
```

Comment:
setters/getters are non idiomatic in go 
```suggestion
	IsRetryableErr func(err error) bool
```